sdk/search/azure-search/samples/async_samples/sample_facet_query_async.py

```diff
@@ -0,0 +1,54 @@
+# coding: utf-8
+
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# --------------------------------------------------------------------------
+
+"""
+FILE: sample_filter_query_async.py
+DESCRIPTION:
+    This sample demonstrates how to authenticate with the Azure Congnitive Search
+    service with an API key. See more details about authentication here:
+    https://docs.microsoft.com/en-us/azure/search/search-security-api-keys
+USAGE:
+    python sample_filter_query_async.py
+
+    Set the environment variables with your own values before running the sample:
+    1) AZURE_SEARCH_SERVICE_ENDPOINT - the endpoint of your Azure Cognitive Search service
+    2) AZURE_SEARCH_INDEX_NAME - the name of your search index (e.g. "hotels-sample-index")
+    3) AZURE_SEARCH_API_KEY - your search API key
+"""
+
+import os
+import asyncio
+
+
+service_endpoint = os.getenv("AZURE_SEARCH_SERVICE_ENDPOINT")
+index_name = os.getenv("AZURE_SEARCH_INDEX_NAME")
+key = os.getenv("AZURE_SEARCH_API_KEY")
+
+async def filter_query():
+    # [START facet_query_async]
+    from azure.search.aio import SearchIndexClient
+    from azure.search import SearchApiKeyCredential, SearchQuery
+
+    search_client = SearchIndexClient(service_endpoint, index_name, SearchApiKeyCredential(key))
+
+    query = SearchQuery(search_text="WiFi", facets=["Category"])
```

Comment:
nit: If the sample is supposed to show getting *only* facets, then typically the app would pass top=0.

Reply:
updated in https://github.com/Azure/azure-sdk-for-python/pull/10207/commits/81e88769a8b010142e68e55ba187e319e9b49947